### ISSUE TITLE
How to collect or cancel a reserved payment?

### ISSUE THREAD

**Issue by mikepence** [tylerhunt/remit#14]
I have been struggling with this for an embarrassingly long time. Once I use a single use pipeline to reserve a payment, what do I do to get a URL to collect or to cancel that reserved payment?

**Comment by pboling** [comment]
I am not sure I completely follow what you mean.  However, in the system I worked on we _reserved_ payments with the single use pipeline, and Amazon would send the results back to a URL we specified in our Amazon FPS settings (on the Amazon website).

Setting up Amazon FPS is EXTREMELY frustrating, and it is a terrible system. :/  Amazon's developer forums are pretty good though, and are monitored by actual Amazon FPS devs.

As to cancelling (prior to payment), I have no idea if this is even possible.  Refunding is possible.

**Comment by mikepence** [comment]
Thanks for taking the time to reply, Peter. I am able to reserve the
payment just fine -- it is when it is time to settle/collect that
payment that I am trying to figure out.

On Tue, Nov 29, 2011 at 3:54 PM, Peter Boling
reply@reply.github.com
wrote:

**Comment by pboling** [comment]
I must not have used remit in the way you are attempting to.  If Amazon has a 'reserve' payment that is never collected until you send a request to collect it I am unaware of it.  But that doesn't mean it isn't there.  The API is enormous, and I only used Remit for one tiny slice of it which included payments which were automatically charged by Amazon, with no (little) waiting.  There may be other remit users who have experience with this.  I'll leave this issue open.  Please report back if you gain any insight on the Amazon dev forums.

**Comment by roysim** [comment]
@mikepence, have you tried calling settle?
Pass in the reserve_transaction_id you got back from Amazon when you called reserve.  And pass in all of the other transaction details you passed to Amazon in reserve.

```
settle_request = Remit::Settle::Request.new(:reserve_transaction_id => reserve_transactionID,...)
@remit_api.settle(settle_request)
```

**Comment by mikepence** [comment]
I am setting up a single use pipeline with :reserve => true, and what I get back includes a tokenID, but no reserve_transaction_id. Maybe I need to be doing it differently...

**Comment by roysim** [comment]
You need to take that token from the pipeline and use it to call Settle.  The pipeline only authorizes the transaction from the user.  You still need to tell the API to execute it.

**Comment by mikepence** [comment]
Right, I think I have got it. After the pipeline returns a token, I have to call Reserve with that token, then, subsequently call Settle. The docs are plentiful, but very obtuse.

**Comment by mikepence** [comment]
What is odd is that doing a pipeline call with reserve=true has no apparent effect.
